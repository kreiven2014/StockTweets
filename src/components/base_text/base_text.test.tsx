/* REACT */
import React from "react";

/* MODULES */
import renderer from "react-test-renderer";

/* CUSTOM MODULES */
import BaseText from "./base_text";

describe("BaseText tests", () => {
  test("renders correctly", (): void => {
    const tree = renderer
      .create(<BaseText textKey="sign_up:link_btn" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with child", (): void => {
    const tree = renderer.create(<BaseText>Test text</BaseText>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly when title", (): void => {
    const tree = renderer
      .create(<BaseText isTitle textKey="sign_up:link_btn" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
