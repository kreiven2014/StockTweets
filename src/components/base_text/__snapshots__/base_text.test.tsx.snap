// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText tests renders correctly 1`] = `
<Text
  allowFontScaling={false}
  ellipsizeMode="tail"
  numberOfLines={1}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 24,
        "letterSpacing": 0.6,
        "lineHeight": 27.5,
        "textAlign": "left",
      },
      undefined,
      undefined,
    ]
  }
>
  sign_up:link_btn
</Text>
`;

exports[`BaseText tests renders correctly when title 1`] = `
<Text
  allowFontScaling={false}
  ellipsizeMode="tail"
  numberOfLines={1}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 24,
        "letterSpacing": 0.6,
        "lineHeight": 27.5,
        "textAlign": "left",
      },
      Object {
        "fontSize": 31,
        "fontWeight": "900",
        "lineHeight": 37.5,
      },
      undefined,
    ]
  }
>
  sign_up:link_btn
</Text>
`;

exports[`BaseText tests renders correctly with child 1`] = `
<Text
  allowFontScaling={false}
  ellipsizeMode="tail"
  numberOfLines={1}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 24,
        "letterSpacing": 0.6,
        "lineHeight": 27.5,
        "textAlign": "left",
      },
      undefined,
      undefined,
    ]
  }
>
  Test text
</Text>
`;
